tfjs-backend-webgpu/src/binary_op_util.ts

```diff
@@ -201,26 +178,54 @@ const PRELU_VEC4 = `
 const SQUARED_DIFFERENCE = 'return (a - b) * (a - b);';
 const SUB = 'return a - b;';
 
-function getBinaryWithNanString(op: string, useVec4: boolean) {
-  const checkNanSnippet = useVec4 ? CHECK_NAN_SNIPPET_VEC4 : CHECK_NAN_SNIPPET;
-  return useVec4 ? `
-    var resultTemp = vec4<f32>(${op}(a, b));
-    ` + checkNanSnippet +
-          `
-    return resultTemp;
-  ` :
-                   checkNanSnippet + `
-    return ${op}(a, b);
-  `;
-}
+const ATAN2 = 'var resultTemp = atan2(a, b);';
+const MAX = 'var resultTemp = max(a, b);';
+const MIN = 'var resultTemp = min(a, b);';
 
 export function getBinaryOpString(
     type: BinaryOpType, useVec4?: boolean): string {
+  // NaN-sensitive ops
+  do {
+    let doOpSnippet: string;
+    switch (type) {
+      case BinaryOpType.ATAN2:
+        doOpSnippet = ATAN2;
+        break;
+      case BinaryOpType.MAX:
+        doOpSnippet = MAX;
+        break;
+      case BinaryOpType.MIN:
+        doOpSnippet = MIN;
+        break;
+      case BinaryOpType.MOD:
+        doOpSnippet = useVec4 ? MOD_VEC4 : MOD;
+        break;
+      case BinaryOpType.NOT_EQUAL:
+        doOpSnippet = useVec4 ? NOT_EQUAL_VEC4 : NOT_EQUAL;
+        break;
+      case BinaryOpType.POW:
+        doOpSnippet = useVec4 ? POW_VEC4 : POW;
+        break;
+      default:
+        continue;
+    }
+    const checkNaNSnippet =
+        useVec4 ? CHECK_NAN_SNIPPET_VEC4 : CHECK_NAN_SNIPPET;
```

Comment:
This is a bit strange to have a new variable whose name is the same as one of conditions. Can we just use "useVec4 ? CHECK_NAN_SNIPPET_VEC4 : CHECK_NAN_SNIPPET" directly?

Reply:
While it's not the case here, sometimes clang-format behaves poorly on `${expression}`. To reiterate [my previous point](https://github.com/tensorflow/tfjs/pull/7193#discussion_r1053949182), personally I prefer the `${identifier}` style, or macro expansion in terms of C/C++.

Reply:
clang-format behaves poorly on ${}, no matter it's expression or identifier. We can now leave it as is.